### DISCUSSION TITLE
What is the use of client.simGetCollisionInfo().object_id in detecting collisions?

### DISCUSSION THREAD

**Discussion by Deepakgthomas** [microsoft/AirSim#3745] [Support Q&A]
More specifically, what's the difference between client.simGetCollisionInfo().object_id!=-1 and client.simGetCollisionInfo().has_collided?

@jonyMarino Could you please help me with this? Thank you!

**Comment by Deepakgthomas** [comment]
@rajat2004 Can you help with this? I hope you don't mind me tagging you. Please let me know if that's the case.

**Comment by jonyMarino** [comment]
Hi @Deepakgthomas! There is no difference, so client.simGetCollisionInfo().has_collided is preferable.
If you want to know with which object you have collided, then use client.simGetCollisionInfo().object_id

**Reply by Deepakgthomas** [reply]
Thank you for the quick reply @jonyMarino ! Based on my experience, I have found that checking client.simGetCollisionInfo().object_id!=-1 is more sensitive as compared to  client.simGetCollisionInfo().has_collided. Basically, the latter doesn't seem to detect collisions sometimes, while that is not the case with the former. I could do further experiments to explain the issue sometime later.

**Reply by jonyMarino** [reply]
Thanks, @Deepakgthomas! I didn't know that. Yes, it would be great if you can trace that in our code. We need more airsim-code experts in the community.
